code/client/scripts/index.js

```diff
@@ -1224,7 +1224,7 @@ function ShareLink(channel, entity, product_imgurl, product_name, product_brand)
                     pin_desc = pin_desc.trim().slice(0, -1)
                 }
             }
-            window.open('https://pinterest.com/pin/create/link/?url=' + encodeURIComponent(url) + '&media=' + encodeURIComponent(product_imgurl) + '&description=' + escape(pin_desc), "_blank");
+            window.open('https://pinterest.com/pin/create/link/?url=' + encodeURIComponent(url.replace("https","http")) + '&media=' + encodeURIComponent(product_imgurl) + '&description=' + escape(pin_desc), "_blank");
```

Comment:
I did a bit of debugging and it appears that the `url` variable at this point in the code is `https%3A//bagcupid.com/app/tags/3`.  I'm not sure where the `url` variable is sourced from, but it is already partially encoded and it shouldn't be.  You will need to figure out what is doing the partial encoding and stop it to fix this bug.